blogpost: featherless as a provider (#2883)
huggingface/blog: inference-providers-feeatherless.md

---
title: "Featherless on Hugging Face Inference Providers 🔥"
thumbnail: /blog/assets/inference-providers-featherless/thumbnail.png
authors:
- user: wxgeorge
  guest: true
  org: featherless-ai
- user: pohnean-recursal
  guest: true
  org: featherless-ai
- user: picocreator
  guest: true
  org: featherless-ai
- user: celinah
- user: sbrandeis
---

![banner image](https://place-hold.it/1680x900)
<!-- ![banner image](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/featherless-banner.png) -->
<!-- TODO: add a banner -->

# Featherless on Hugging Face Inference Providers 🔥

> We're thrilled to share that **Featherless** is now a supported Inference Provider on HF Hub!

Reviewer: ```suggestion
We're thrilled to share that **Featherless** is now a supported Inference Provider on the Hugging Face Hub!
```

Featherless joins our growing ecosystem, enhancing the breadth and capabilities of serverless inference directly on the Hub’s model pages. Inference Providers are also seamlessly integrated into our client SDKs (for both JS and Python), making it super easy to use a wide variety of models with your preferred providers.

[Featherless](https://featherless.ai) supports a wide variety of text and conversational models, including the latest open-source models from DeepSeek, Meta, Google, Qwen, and much more.

> Find the full list of supported models on the [models page](https://huggingface.co/models?inference_provider=featherless-ai&sort=trending).

Reviewer: ```suggestion
Featherless is a serverless AI inference provider with unique model loading and GPU orchestration abilities that makes an exceptionally large catalog of models available for users. Providers often offer either a low cost of access to a limited set of models, or an unlimited range of models with users managing servers and the associated costs of operation. Featherless provides the best of both worlds offering unmatched model range and variety but with serverless pricing. Find the full list of supported models on the [models page](https://huggingface.co/models?inference_provider=featherless-ai&sort=trending).
```

Taken from the "Why Featherless" section from https://featherless.ai/. Let's mention their specificity which is to provide access to 1000+ models at once.

We're quite excited to see what you'll build with this new provider!

> Read more about Inference Providers in our [documentation](https://huggingface.co/docs/inference-providers).

Reviewer: ```suggestion
Read more about how to use Featherless as Inference Provider in its dedicated [documentation page](https://huggingface.co/docs/inference-providers/providers/featherless-ai).
```

We can redirect to the specific doc page. Requires https://github.com/huggingface/hub-docs/pull/1748 to be merged (let's do it now so it'd done? PR is ready to be merge rn cc @hanouticelina)

2. As mentioned, there are two modes when calling Inference Providers: 
- Custom key (calls go directly to the inference provider, using your own API key of the corresponding inference provider)
- Routed by HF (in that case, you don't need a token from the provider, and the charges are applied directly to your HF account rather than the provider's account)

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/explainer.png" alt="Inference Providers"/>

3. Model pages showcase third-party inference providers (the ones that are compatible with the current model, sorted by user preference)

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/model-widget-updated.png" alt="Inference Providers"/>

### From the client SDKs

#### from Python, using huggingface_hub

The following example shows how to use DeepSeek-R1 using Hyperbolic as the inference provider. You can use a [Hugging Face token](https://huggingface.co/settings/tokens) for automatic routing through Hugging Face, or your own Hyperbolic API key if you have one.

> Install `huggingface_hub` from source (see [instructions](https://huggingface.co/docs/huggingface_hub/installation#install-from-source)). Official support will be released soon in version v0.29.0.

Reviewer 2: ```suggestion
Install `huggingface_hub` from source (see [instructions](https://huggingface.co/docs/huggingface_hub/installation#install-from-source)). Official support will be released soon in version v0.33.0.
```

```python
from huggingface_hub import InferenceClient

client = InferenceClient(
    provider="featherless-ai",
    api_key="xxxxxxxxxxxxxxxxxxxxxxxx"
)

messages = [
    {
        "role": "user",
        "content": "What is the capital of France?"
    }
]

completion = client.chat.completions.create(
    model="deepseek-ai/DeepSeek-R1-0528", 
    messages=messages, 

>     max_tokens=500

Reviewer: ```suggestion
```

(nit) not really useful in the end so it has been removed from doc snippets

)

print(completion.choices[0].message)
```

#### from JS using @huggingface/inference

```js
import { HfInference } from "@huggingface/inference";

const client = new HfInference("xxxxxxxxxxxxxxxxxxxxxxxx");

const chatCompletion = await client.chatCompletion({
	model: "deepseek-ai/DeepSeek-R1-0528",
	messages: [
		{
			role: "user",
			content: "What is the capital of France?"
		}
	],
	provider: "featherless-ai",

> 	max_tokens: 500

Reviewer: ```suggestion
```

(same)
